File: app/schemas/alert.py
from datetime import datetime
from pydantic import BaseModel
from app.db.enums import AlertStatus, Severity

class AlertUpdateRequest(BaseModel):
    status: AlertStatus

class AlertResponse(BaseModel):
    id: int
    event_id: int
    title: str
    severity: Severity
    status: AlertStatus
    risk_score: int
    description: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/alert.ts
export type AlertResponse = {
  id: number;
  event_id: number | null;
  title: string;
  severity: string;
  status: string;
  risk_score: number;
  description: string | null;
  created_at: string;
  updated_at: string | null;
};

export type AlertExplanationResponse = {
  alert_id: number;
  title: string;
  severity: string;
  risk_score: number;
  status: string;
  summary: string;
  severity_explanation: string;
  possible_causes: string[];
  recommended_actions: string[];
  source: string;
};